packages/idempotency/src/makeFunctionIdempotent.ts

```diff
@@ -0,0 +1,9 @@
+import { AnyFunction } from 'types/AnyFunction';
+import { IdempotencyOptions } from './IdempotencyOptions';
+
+const makeFunctionIdempotent = (
```

Comment:
Should we make this a type as well that way we can pass the generic from here down. So we makeFunctionIdempotent<U> and the function would return that Type? Or am I missing something? Otherwise isnt the generic not propagated?